Lib/ipaddress.py

```diff
@@ -1287,7 +1287,7 @@ def __init__(self, address):
             return
 
         # Constructing from a packed address
-        if isinstance(address, bytes):
+        if isinstance(address, bytes) or isinstance(address, bytearray):
```

Comment:
I think it would be better to pass a tuple to isinstance.
```python
if isinstance(address, (bytes, bytearray)):
```

And there are couple of more code lines that use `isinstance`